server/app/views/components/TextFormatter.java

```diff
@@ -1,190 +1,121 @@
-        builder.add(buildAccordion(accordionHeader, accordionContent));
-      } else if (line.startsWith(TextFormatter.BULLETED_ITEM)) { // unordered list item.
-        ArrayList<String> items = new ArrayList<>();
-        items.add(line.substring(1).trim());
-        int next = i + 1;
-        while (next < lines.length && lines[next].startsWith(TextFormatter.BULLETED_ITEM)) {
-          items.add(lines[next].substring(1).trim());
-          next++;
-        }
-        i = next - 1;
-        builder.add(buildList(items));
-      } else if (line.length() > 0) {
-        ImmutableList<DomContent> lineContent =
-            TextFormatter.createLinksAndEscapeText(
-                line, UrlOpenAction.NewTab, /*addRequiredIndicator= */ false);
-        builder.add(div().with(lineContent));
-      } else if (preserveEmptyLines) {
-        builder.add(div().withClasses("h-6"));
+      if (line.isBlank()) {
+        // Add an empty space character so the blank line is preserved
+        lines[i] = "&nbsp;\n";
       }
     }
-    return builder.build();
+    return String.join("\n", lines);
   }
 
-  private static DivTag buildAccordion(String title, String accordionContent) {
-    Accordion accordion = new Accordion().setTitle(title);
-    ImmutableList<DomContent> contentTags = TextFormatter.formatText(accordionContent, true);
-    contentTags.forEach(accordion::addContent);
-    return accordion.getContainer();
+  private static String addIconToLinks(String markdownText) {
+    String closingATag = "</a>";
+    String svgIconString =
+        Icons.svg(Icons.OPEN_IN_NEW)
+            .withClasses("shrink-0", "h-5", "w-auto", "inline", "ml-1", "align-text-top")
+            .toString();
+    return markdownText.replaceAll(closingATag, svgIconString + closingATag);
   }
 
-  private static UlTag buildList(List<String> items) {
-    UlTag listTag = ul().withClasses("list-disc", "mx-8");
-    items.forEach(item -> listTag.with(li().withText(item)));
-    return listTag;
+  private static String addRequiredIndicatorInsidePTag(String markdownText) {
+    int indexOfPTag = markdownText.lastIndexOf("</p>");
+    String stringWithRequiredIndicator = ViewUtils.requiredQuestionIndicator().toString() + "</p>";
+    return markdownText.substring(0, indexOfPTag) + stringWithRequiredIndicator;
+  }
+
+  private static String sanitizeHtml(String markdownText) {
+    PolicyFactory customPolicy =
+        new HtmlPolicyBuilder()
+            .allowElements(
+                "p", "div", "h2", "h3", "h4", "h5", "h6", "a", "ul", "li", "span", "svg", "br",
+                "em", "strong", "code", "path", "pre")
+            // Per accessibility best practices, we want to disallow adding h1 headers to
+            // ensure the page does not have more than one h1 header
+            // https://www.a11yproject.com/posts/how-to-accessible-heading-structure/
+            // This logic changes h1 headers to h2 headers which are still larger than the default
```

Comment:
If we're changing h1s to h2s, should we also change h2s to h3s so that there is a relative difference in size?  Otherwise, a user may not understand why "#" and "##" produce the same size headers.

Reply:
I'm torn about this. My instinct is to not change more than we need to, but I agree it could be confusing. I'll think on it some more!

Reply:
I'm leaning towards leaving this as is. In our communication we can specify that h1 headers aren't allowed and direct people to use h2 headers which should clear up confusion. Otherwise we would have to change all of the smaller headers as well. What do you think?

Reply:
That makes sense to me, especially since the only header we support right now is h2.